convokit/storage/convoKitIndex.py

```diff
@@ -0,0 +1,270 @@
+        return self.fields.__getitem__('utterances_index')
+
+    @utterances_index.setter
+    def utterances_index(self, new_utterances_index):
+        self.fields.__setitem__('utterances_index', new_utterances_index)
+
+    @property
+    def speakers_index(self):
+        return self.fields.__getitem__('speakers_index')
+
+    @speakers_index.setter
+    def speakers_index(self, new_speakers_index):
+        self.fields.__setitem__('speakers_index', new_speakers_index)
+
+    @property
+    def conversations_index(self):
+        return self.fields.__getitem__('conversations_index')
+
+    @conversations_index.setter
+    def conversations_index(self, new_conversations_index):
+        self.fields.__setitem__('conversations_index', new_conversations_index)
+
+    @property
+    def overall_index(self):
+        return self.fields.__getitem__('overall_index')
+
+    @overall_index.setter
+    def overall_index(self, new_overall_index):
+        self.fields.__setitem__('overall_index', new_overall_index)
+
+    @property
+    def vectors(self):
+        return set(self.fields.__getitem__('vectors'))
+
+    @vectors.setter
+    def vectors(self, new_vectors):
+        self.fields.__setitem__('vectors', list(new_vectors))
+
+    @property
+    def version(self):
+        return self.fields.__getitem__('version')
+
+    @version.setter
+    def version(self, new_version):
+        self.fields.__setitem__('version', new_version)
+
+    @property
+    def type_check(self):
+        return self.fields.__getitem__('type_check')
+
+    @type_check.setter
+    def type_check(self, new_type_check):
+        self.fields.__setitem__('type_check', new_type_check)
+
+    @property
+    def lock_metadata_deletion(self):
+        return self.fields.__getitem__('lock_metadata_deletion')
+
+    @lock_metadata_deletion.setter
+    def lock_metadata_deletion(self, new_lock_metadata_deletion):
+        self.fields.__setitem__('lock_metadata_deletion',
+                                new_lock_metadata_deletion)
+
+    ############
+
+    def update_index(self, obj_type: str, key: str, class_type: str):
+        """
+        Append the class_type to the index
+
+        :param obj_type: utterance, conversation, or speaker
+        :param key: string
+        :param class_type: class type
+        :return: None
+        """
+        assert type(key) == str
+        assert 'class' in class_type or class_type == 'bin'
+        if key not in self.indices[obj_type]:
+            self.indices[obj_type][key] = []
+        if class_type not in self.indices[obj_type][key]:
+            self.indices[obj_type][key].append(class_type)
+
+    def set_index(self, obj_type: str, key: str, class_type: str):
+        """
+        Set the class_type of the index as [`class_type`].
+
+        :param obj_type: utterance, conversation, or speaker
+        :param key: string
+        :param class_type: class type
+        :return: None
+        """
+        assert type(key) == str
```

Comment:
We should avoid the use of `assert` outside of test scripts, since failed assertions do not print a human-readable error message. We should instead raise an exception:
```suggestion
        if type(key) != str:
            raise TypeError(f'key must be a string (got {type(key)} instead)')
```
(And likewise for all other uses of `assert` throughout the file)